app/src/main/java/eu/kanade/tachiyomi/data/backup/restore/restorers/ExtensionRepoRestorer.kt

```diff
@@ -0,0 +1,37 @@
+package eu.kanade.tachiyomi.data.backup.restore.restorers
+
+import eu.kanade.tachiyomi.data.backup.models.BackupExtensionRepos
+import mihon.domain.extensionrepo.interactor.GetExtensionRepo
+import tachiyomi.data.DatabaseHandler
+import uy.kohesive.injekt.Injekt
+import uy.kohesive.injekt.api.get
+
+class ExtensionRepoRestorer(
+    private val handler: DatabaseHandler = Injekt.get(),
+    private val getExtensionRepos: GetExtensionRepo = Injekt.get()
+) {
+
+    suspend operator fun invoke(backupExtensionRepos: List<BackupExtensionRepos>) {
+        if (backupExtensionRepos.isNotEmpty()) {
+            val dbExtensionRepos = getExtensionRepos.getAll()
+            val dbExtensionReposByName = dbExtensionRepos.associateBy { it.name }
+
+            backupExtensionRepos
+                .sortedBy { it.signingKeyFingerprint }
```

Comment:
Why?

Reply:
Are you are talking about how I used the name to check if it already exists, if so, my bad, just forgot to switch it after copying it from another file. I've changed it to use the SHA-256 for the comparisons, which makes infinitely more sense.